poolscript/script.go

```diff
@@ -498,6 +500,22 @@ func IsTaprootExpirySpend(witness wire.TxWitness) bool {
 	return true
 }
 
+// hasAnnex returns true if the provided witness includes an annex element,
+// otherwise returns false.
+func hasAnnex(witness wire.TxWitness) bool {
+	// By definition, the annex element can not be the sole element in the witness stack.
+	if len(witness) < 2 {
+		return false
+	}
```

Comment:
nit: we usually separate the blocks with empty lines, this function would be then three brlocks (two breaks)